ee/vellum_ee/workflows/display/workflows/base_workflow_display.py

```diff
@@ -794,6 +801,98 @@ def _serialize_integration_trigger_exec_config(self, trigger_class: Type[Integra
+                f"v1/integration-providers/{provider}/triggers",
+                base_url=self._client.integration_providers._client_wrapper.get_environment().default,
+                method="GET",
+                params={"integration_name": integration_name},
+            )
+            if response.status_code != 200:
+                logger.warning(
+                    f"Failed to fetch trigger definitions for {provider}/{integration_name}: {response.status_code}"
+                )
+                return None
+
+            data = response.json()
+            results = data.get("results", [])
+            for trigger_def in results:
+                if trigger_def.get("name") == trigger_slug:
+                    return cast(JsonObject, trigger_def)
+
+            logger.warning(f"Trigger {trigger_slug} not found in {provider}/{integration_name}")
+            return None
+        except Exception as e:
+            logger.warning(f"Error fetching trigger definition: {e}")
+            return None
+
+    def _validate_integration_trigger_attributes(
+        self,
+        trigger_class: Type[IntegrationTrigger],
+        trigger_attributes: JsonArray,
+    ) -> None:
+        """
+        Validate that the trigger attributes match the expected types from the API.
+
+        Raises TriggerValidationError if there's a type mismatch.
+        """
+        config_class = trigger_class.Config
+        provider = getattr(config_class, "provider", None)
+        if isinstance(provider, Enum):
+            provider = provider.value
+        elif provider is not None:
+            provider = str(provider)
+
+        slug = getattr(config_class, "slug", None)
+        integration_name = getattr(config_class, "integration_name", None)
+
+        if not provider or not slug or not integration_name:
+            return
+
+        trigger_def = self._fetch_integration_trigger_definition(provider, integration_name, slug)
+        if not trigger_def:
+            return
+
+        expected_event_attributes = trigger_def.get("event_attributes", [])
+        if not expected_event_attributes or not isinstance(expected_event_attributes, list):
+            return
+
+        expected_types_by_key: Dict[str, str] = {}
+        for attr in expected_event_attributes:
+            if not isinstance(attr, dict):
+                continue
+            key = attr.get("key")
+            attr_type = attr.get("type")
+            if isinstance(key, str) and isinstance(attr_type, str):
+                expected_types_by_key[key] = attr_type
+
+        for attr in trigger_attributes:
+            if not isinstance(attr, dict):
+                continue
+            key = attr.get("key")
+            actual_type = attr.get("type")
+            if isinstance(key, str) and key in expected_types_by_key:
+                expected_type = expected_types_by_key[key]
+                if actual_type != expected_type:
```

Comment:
[q] Is this correct? I would have expected one to be a json schema and the other to be a python type